src/lib/mina_net2/libp2p_stream.ml

```diff
@@ -126,6 +126,40 @@ let stream_closed ~logger ~who_closed t =
         ] ;
   `Stream_should_be_released (equal_state FullyOpen t.state)
 
+let max_message_size = 16777216 (* 16 MiB *)
+
+let split_string ~every b =
+  let blen = String.length b in
```

Comment:
Nit: this would be faster with `Array.init`, especially for large messages

Reply:
I used `List.init`, please check what's now